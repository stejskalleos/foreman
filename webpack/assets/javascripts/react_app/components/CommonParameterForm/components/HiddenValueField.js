import React from 'react';
import PropTypes from 'prop-types';

import { translate as __ } from '../../../common/I18n';
import FormField from '../../common/forms/FormField';

const HiddenValueField = ({ value, onChange }) => (
  <FormField label={__('Hidden Value')}>
    <input
      type="checkbox"
      id="common_parameter_hidden_value"
      name="common_parameter[hidden_value]"
      checked={value}
      onChange={() => onChange(!value)}
    />
  </FormField>
);

HiddenValueField.propTypes = {
  value: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
};

HiddenValueField.defaultProps = {
  value: false,
};

export default HiddenValueField;
