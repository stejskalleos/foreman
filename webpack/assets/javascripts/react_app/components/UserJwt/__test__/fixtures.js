import { noop } from '../../../common/helpers';
import { STATUS } from '../../../constants';

export const formProps = {
  handleSubmit: noop,
};

const token = 'token23';

export const fieldOk = {
  apiStatus: STATUS.RESOLVED,
  token,
};

export const fieldError = {
  apiStatus: STATUS.ERROR,
  token,
};

export const fieldPending = {
  apiStatus: STATUS.PENDING,
  token,
};
