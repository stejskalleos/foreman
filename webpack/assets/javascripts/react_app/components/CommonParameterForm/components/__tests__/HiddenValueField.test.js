import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { hiddenValueFieldFixtures } from './fixtures';
import HiddenValueField from '../HiddenValueField';

describe('HiddenValueField', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(HiddenValueField, hiddenValueFieldFixtures));
});
