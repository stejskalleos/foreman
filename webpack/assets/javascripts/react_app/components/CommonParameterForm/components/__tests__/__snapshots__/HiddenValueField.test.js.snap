// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiddenValueField rendering when displayed 1`] = `
<FormField
  className=""
  disabled={false}
  error={null}
  helpInline={null}
  id={null}
  inputProps={null}
  inputSizeClass="col-md-4"
  label="Hidden Value"
  labelHelp={null}
  labelSizeClass="col-md-2"
  onChange={[Function]}
  required={false}
  type="text"
>
  <input
    checked={false}
    id="common_parameter_hidden_value"
    name="common_parameter[hidden_value]"
    onChange={[Function]}
    type="checkbox"
  />
</FormField>
`;

exports[`HiddenValueField rendering when hidden 1`] = `
<FormField
  className=""
  disabled={false}
  error={null}
  helpInline={null}
  id={null}
  inputProps={null}
  inputSizeClass="col-md-4"
  label="Hidden Value"
  labelHelp={null}
  labelSizeClass="col-md-2"
  onChange={[Function]}
  required={false}
  type="text"
>
  <input
    checked={true}
    id="common_parameter_hidden_value"
    name="common_parameter[hidden_value]"
    onChange={[Function]}
    type="checkbox"
  />
</FormField>
`;
